### PR TITLE
msftidy - also check https

### PR DESCRIPTION
**Pull request by firefart**
Also check for https urls

### CODE REVIEW

**Comment by bcoles** [comment]
These modules should probably also be updated in this PR:


```
# grep -rE 'https?:\/\/www\.securityfocus\.com\/bid\/' modules/
modules/exploits/windows/misc/hp_imc_dbman_restartdb_unauth_rce.rb:          ['URL', 'https://www.securityfocus.com/bid/98469/info'],
modules/exploits/windows/misc/hp_imc_dbman_restoredbase_unauth_rce.rb:          ['URL', 'https://www.securityfocus.com/bid/98469/info'],
```

```
# grep -rE 'https?:\/\/www\.kb\.cert\.org\/vuls\/id' modules/
modules/exploits/linux/http/netgear_r7000_cgibin_exec.rb:          [ 'URL', 'https://www.kb.cert.org/vuls/id/582384'],
```

And it looks like I missed a few ExploitDB references in the DoS modules:

```
# grep -rE 'https?:\/\/www\.exploit\-db\.com\/exploits' modules/
# ... snip ...
modules/auxiliary/dos/http/slowloris.py:        {'type': 'url', 'ref': 'https://www.exploit-db.com/exploits/8976/'},
modules/auxiliary/dos/scada/siemens_siprotec4.rb:              [ 'URL', 'https://www.exploit-db.com/exploits/44103/' ],
modules/auxiliary/dos/tcp/claymore_dos.py:        {'type': 'url', 'ref': 'https://www.exploit-db.com/exploits/43972/'},
# ... snip ...
```

**Comment by FireFart** [comment]
I just encountered a funny thing. My msftidy did not complain about anything because of an edge case:
I have the git folder mounted on virtualbox from my windows machine. It looks like virtualbox mounts everything with execute permissions:
```
[firefart@fedora metasploit-framework]$ ls -alh modules/exploits/windows/misc/hp_imc_dbman_restartdb_unauth_rce.rb
-rwxrwx---. 1 root vboxsf 4,1K 27. Aug 21:31 modules/exploits/windows/misc/hp_imc_dbman_restartdb_unauth_rce.rb
```

But this causes this line inside msftidy to skip every module:
```
next if File.executable?(full_filepath)
```

What do you think is it safe to drop the executable check too? One line before we check for a .rb extension so this only applies to executable ruby scripts inside metasploit.

**Comment by FireFart** [comment]
Jeah but I haven't found one based on the filename (the checks all happen before reading the content). So I think the only way would be to move the check to a point where we already read the content and check for "Metasploit.run" or "require 'metasploit'". What do you think?

**Comment by wvu-r7** [comment]
That sounds reasonable to me.

**Comment by wvu-r7** [comment]
Paging Dr. @acammack-r7 for his input.

**Comment by FireFart** [comment]
I changed the check a little and now it also works on mounted volumes

File: tools/dev/msftidy.rb [hunk 1]
```diff
@@ -745,10 +745,11 @@ def cleanup_text(txt)
...
         next if full_filepath =~ /\.git[\x5c\x2f]/
         next unless File.file? full_filepath
-        next unless full_filepath =~ /\.rb$/
-        # Executable files are now assumed to be external modules
-        next if File.executable?(full_filepath)
+        next unless File.extname(full_filepath) == '.rb'
         msftidy = Msftidy.new(full_filepath)
+        # Executable files are now assumed to be external modules
+        # but also check for some content to be sure
+        next if File.executable?(full_filepath) && msftidy.source.include?("require 'metasploit'")
```
Comment (wvu-r7) [inline tools/dev/msftidy.rb]: I'd check for double quotes, too.

Comment (FireFart) [reply tools/dev/msftidy.rb]: done

**Comment by acammack-r7** [comment]
This revised external Ruby module check makes sense. Thanks for the catch!

**Comment by wvu-r7** [comment]
```
wvu@kharak:~/metasploit-framework:master$ tools/dev/msftidy.rb modules/auxiliary/dos/smb/smb_loris.rb
wvu@kharak:~/metasploit-framework:master$
```

**Comment by wvu-r7** [comment]
Post-merge:

```
wvu@kharak:~/metasploit-framework:master$ tools/dev/msftidy.rb modules/auxiliary/dos/smb/smb_loris.rb
wvu@kharak:~/metasploit-framework:master$
```

**Pull request merged**